fmriprep/utils/bids.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Utilities to handle BIDS inputs
+"""
+import os.path as op
+import warnings
+from glob import glob
+from copy import deepcopy
+from bids.grabbids import BIDSLayout
+
+INPUTS_SPEC = {'fieldmaps': [], 'func': [], 't1': [], 'sbref': [], 't2w': []}
+
+
+def collect_participants(bids_dir, participant_label=None):
+    """
+    Lists the participants under the BIDS root and checks that participants
+    designated with the participant_label argument exist in that folder.
+
+    Returns the list of participants to be finally processed.
+
+
+    """
+    bids_dir = op.abspath(bids_dir)
+    all_participants = [op.basename(subdir)[4:]
+                        for subdir in glob(op.join(bids_dir, 'sub-*'))]
```

Comment:
Not a big deal to change, but `glob` seems a bit heavy for this job:

```Python
all_participants = [subdir[4:] for subdir in os.listdir(bids_dir) if subdir.startswith('sub-')]
```

Could also add an `op.isdir` check, if there's any concern there.